bubble_sort.rb

```diff
@@ -0,0 +1,55 @@
+def bubble_sort(array)
+  is_sorted = false
+
+  i = 0
+  length = array.length - 1
+
+  loop do
+    j = 0
+    is_sorted = true
+
+    break if i == length
+
+    loop do
+      break if j + i == length
+
+      if array[j] > array[j + 1]
+        array[j], array[j + 1] = array[j + 1], array[j]
+        is_sorted = false
+      end
+      break if j == length - 1
+
+      j += 1
+    end
+    break if is_sorted
+
+    i += 1
+  end
+  array.to_s
+end
+
+def bubble_sort_by(array)
+  return 'No block given' unless block_given?
```

Comment:
Ruby has a default error that is raised when an expected block is not provided. Let's remove this error handling and allow the default error.

Reply:
**Noted** I put that there as a fail-safe :safety_pin: